src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -151,6 +152,11 @@ export interface SavedObjectsServiceSetup {
    * ```
    */
   registerType: (type: SavedObjectsType) => void;
+
+  /**
+   * TODO: documentation
+   */
+  registerExportHook: (type: string, exportHook: SavedObjectsTypeExportHook) => void;
```

Comment:
As discussed in https://github.com/elastic/kibana/issues/84980#issuecomment-756239433, The hooks are registered via the `setup` contract, but using a distinct API instead of being included in the type when registering them via `registerType`. See the comment on #84980 for the reasoning behind that decision.

Reply:
Note: after discussion starting at https://github.com/elastic/kibana/issues/84980#issuecomment-759473258, `registerExportHook` will be removed in favor of registering the transform via `SavedObjectType.onExport`